### DISCUSSION TITLE
How to navigate away from code-server, triggered by an extension

### DISCUSSION THREAD

**Discussion by protozoo** [coder/code-server#6079] [General]
Hi there!

I am developing a vscode extension and running it on a code-server instance. The extension is composed of several commands and a couple of webviews. I am now trying to implement a functionality that requires the extension to be able to navigate away from the current location (the url hosting the code-server instance) but I can't find any way of doing so. Since my extension has a couple of webviews, and those are "just" iframes, I tried something like 

window.top.location.href = myNewUrl;

But I get a security error saying that an iframe can not trigger such action.
So my question is, how could I implement such feature? Maybe there's some function or API I could call? Perhaps a listener for specific message using postMessage()?

Thanks in advance

**Comment by code-asher** [comment]
As far as API goes I think the best you can do is open a new tab/window with something like `vscode.env.openExternal(uri)` (untested).

**Reply by protozoo** [reply]
Thanks for the reply, but I need to navigate away in the current tab. The use case is to close the session after a user's inactivity period
